File: PowerFitControl/students/src/main/kotlin/br/com/pwrftctrl/students/domain/model/StudentModel.kt
package br.com.pwrftctrl.students.domain.model

import br.com.pwrftctrl.core.domain.models.UserModel
import java.time.LocalDate

data class StudentModel(
        val id: Int,
        // Dados do aluno
        val name: String,
        val lastName: String,
        val cpf: String,
        val dateOfBirth: LocalDate,
        val email: String,
        val phone1: String,
        val phone2: String,
        val whatsapp: String,
        val gender: String,
        val avaterUrl: String,
        // Endereço
        val city: String,
        val neighborhood: String,
        val street: String,
        val avenue: String,
        val zipCode: String,
        val number: String,
        val reference: String,
        // Matricula
        val startDate: LocalDate,
        val dueDate: LocalDate,
        val modality: Int,
        val observation: String
)

fun StudentModel.toUserModel(): UserModel {
        return UserModel(
                id = this.id,
                cpf = this.cpf,
                firstName = this.name,
                lastName = this.lastName,
                email = this.email,
                dateOfBirth = this.dateOfBirth,
                gender = this.gender,
                phone1 = this.phone1,
                phone2 = this.phone2,
                whatsapp = this.whatsapp,
                photo = this.avaterUrl,
                neighborhood = this.neighborhood,
                zipCode = this.zipCode,
                city = this.city,
        )
}

fun StudentModel.toRegistrationModel(): RegistrationModel {
        return RegistrationModel(
                id = this.id,
                startDate = this.startDate,
                dueDate = this.dueDate,
                modality = this.modality,
                observation = this.observation,
                student = this.id,
        )
}

File: vuetify/academy-system/src/composables/domain/models/StudentModel.ts
import IModelDomain from "./IModelDomain";

export default class implements IModelDomain {
	id?: number;
	active?: boolean;
	createAt?: Date;

	firstName: string;
	lastName: string;
	avatar?: File;
	cpf: string;
	birthday?: Date;
	gender: string;
	mail: string;
	phone: string;
	whatsapp: string;
	usePhone: boolean;
	//
	addressString: string;
	city: string;
	neighborhood: string;
	street: string;
	zipCode: string;
	addressNumber: string;
	reference: string;
	//
	startDate: Date;
	endDate?: Date;
	modality?: number;
	observation: string;
	isMonthlyPlan: boolean;
	constructor(data: {
		active?: boolean;
		firstName: string;
		lastName: string;
		avatar?: File;
		cpf: string;
		birthday?: Date;
		gender: string;
		mail: string;
		phone: string;
		whatsapp: string;
		usePhone: boolean;
		addressString: string;
		city: string;
		neighborhood: string;
		street: string;
		zipCode: string;
		addressNumber: string;
		reference: string;
		startDate?: Date;
		endDate?: Date;
		modality?: number;
		observation: string;
		isMonthlyPlan?: boolean;
		createAt?: Date;
	}) {
		this.active = data.active;
		this.createAt = data.createAt;
		this.firstName = data.firstName;
		this.lastName = data.lastName;
		this.avatar = data.avatar;
		this.cpf = data.cpf;
		this.birthday = data.birthday;
		this.gender = data.gender;
		this.mail = data.mail;
		this.phone = data.phone;
		this.whatsapp = data.whatsapp;
		this.usePhone = data.usePhone;
		this.addressString = data.addressString;
		this.city = data.city;
		this.neighborhood = data.neighborhood;
		this.street = data.street;
		this.zipCode = data.zipCode;
		this.addressNumber = data.addressNumber;
		this.reference = data.reference;
		this.startDate = data.startDate || new Date();
		this.endDate = data.endDate;
		this.modality = data.modality;
		this.observation = data.observation;
		this.isMonthlyPlan = data.isMonthlyPlan || true;
	}
}
